misk-feature-testing/src/main/kotlin/misk/feature/testing/FakeFeatureFlags.kt

```diff
@@ -77,76 +77,171 @@ class FakeFeatureFlags @Inject constructor(private val moshi : Provider<Moshi>)
-  }
+  fun override(
+    feature: Feature,
+    value: Enum<*>
+  ) = override<Enum<*>>(feature, value)
 
-  fun <T> override(feature: Feature, value: T, clazz: Class<T>) {
-    overrides[MapKey(feature)] = { moshi.get().adapter(clazz).toSafeJson(value) }
+  fun <T> override(
+    feature: Feature,
+    value: T
+  ) = overrideKey(feature, KEY, value, defaultAttributes)
+
+  fun <T> override(
+    feature: Feature,
+    value: T,
+    clazz: Class<T>
+  ) {
+    val jsonValue = { moshi.get().adapter(clazz).toSafeJson(value) }
+    overrideKey(feature, KEY, jsonValue, defaultAttributes)
   }
 
   fun overrideJsonString(feature: Feature, json : String) {
-    overrides[MapKey(feature)] = { json }
+    overrideKey(feature, KEY, { json }, defaultAttributes)
   }
 
   inline fun <reified T> overrideJson(feature: Feature, value: T) {
-    override(feature, value, T::class.java)
+    overrideKeyJson(feature, KEY, value, defaultAttributes)
   }
 
-  fun overrideKey(feature: Feature, key: String, value: Boolean) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Boolean,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Boolean>(feature, key, value, attributes)
 
-  fun overrideKey(feature: Feature, key: String, value: Int) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Int,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Int>(feature, key, value, attributes)
 
-  fun overrideKey(feature: Feature, key: String, value: String) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: String,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<String>(feature, key, value, attributes)
 
-  fun overrideKey(feature: Feature, key: String, value: Enum<*>) {
-    overrides[MapKey(feature, key)] = value
-  }
+  fun overrideKey(
+    feature: Feature,
+    key: String,
+    value: Enum<*>,
+    attributes: Attributes = defaultAttributes
+  ) = overrideKey<Enum<*>>(feature, key, value, attributes)
 
-  fun <T> overrideKey(feature: Feature, key: String, value: T, clazz : Class<T>) {
-    overrides[MapKey(feature, key)] = { moshi.get().adapter(clazz).toSafeJson(value) }
+  fun <T> overrideKey(
+    feature: Feature,
+    key: String,
+    value: T,
+    clazz: Class<T>
+  ) {
+    val jsonValue = { moshi.get().adapter(clazz).toSafeJson(value) }
+    overrideKey(feature, key, jsonValue, defaultAttributes)
   }
 
-  inline fun <reified T> overrideKeyJson(feature: Feature, key: String, value: T) {
-    overrideKey(feature, key, value, T::class.java)
+  fun <T> overrideKey(
+    feature: Feature,
+    key: String,
+    value: T,
+    attributes: Attributes = Attributes()
+  ) {
+    val mapKey = MapKey(feature, key)
+    overrides[mapKey] = overrides[mapKey] ?: PriorityQueue()
```

Comment:
I think you want `computeIfAbsent()`

Reply:
nice. I have updated the code accordingly.